cloud/satellite.go

```diff
@@ -120,15 +122,27 @@ func (c *Client) DeleteSatellite(ctx context.Context, name, orgID string) error
 	return nil
 }
 
-func (c *Client) LaunchSatellite(ctx context.Context, name, orgID, platform, size, version, maintenanceWindow string, features []string) error {
+type LaunchSatelliteOpt struct {
+	Name                    string
+	OrgID                   string
+	Size                    string
+	Platform                string
+	PinnedVersion           string
+	MaintenanceWindowStart  string
+	MaintenanceWeekendsOnly bool
+	FeatureFlags            []string
+}
+
+func (c *Client) LaunchSatellite(ctx context.Context, opt LaunchSatelliteOpt) error {
```

Comment:
Stylistic nit: I know they're a lot more effort to implement, but I really prefer [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) over the option struct pattern. They are harder to write, but easier to comprehend and use.

Reply:
Cool, I can try that out in a future PR, since I want to get his merged asap for 0.7 release